transformations/transformations/parallel_routine_dispatch.py

```diff
@@ -11,17 +11,48 @@
     FindNodes, nodes as ir
 )
 from loki.transform import Transformation
+from loki import (
+    FindVariables, DerivedType, SymbolAttributes,
+    Array, single_variable_declaration, Transformer,
+    BasicType
+)
+import pickle
+import os
 
 __all__ = ['ParallelRoutineDispatchTransformation']
 
 
 class ParallelRoutineDispatchTransformation(Transformation):
 
+    def __init__(self):
+        self.horizontal = [
+            "KLON", "YDCPG_OPTS%KLON", "YDGEOMETRY%YRDIM%NPROMA",
+            "KPROMA", "YDDIM%NPROMA", "NPROMA"
+    ]
+        #TODO : do smthg for opening field_index.pkl
+        with open(os.getcwd()+"/transformations/transformations/field_index.pkl", 'rb') as fp:
+            self.map_index = pickle.load(fp)
+        # CALL FIELD_NEW (YL_ZA, UBOUNDS=[KLON, KFLEVG, KGPBLKS], LBOUNDS=[1, 0, 1], PERSISTENT=.TRUE.)
+        self.new_calls = []
+        # IF (ASSOCIATED (YL_ZA)) CALL FIELD_DELETE (YL_ZA)
+        self.delete_calls = []
+        # map[name] = [field_ptr, ptr]
+        # where : 
+        # field_ptr : pointer on field api object
+        # ptr : pointer to the data
+        self.routine_map_temp = {}  
+        self.routine_map_derived = {} 
```

Comment:
Again a remark for later, and I know that this doesn't apply in your use case but would be a requirement for integration into the main branch. So, I'm pointing this out here:
Transformations are intended to be written in a way that allows to instantiate them once and then apply them to an arbitrary number of subroutines. This information is local to one routine but currently it lives on the transformation object. Later, we should consider making these local variables in the `transform_subroutine` and, where necessary, have methods update or return them.